### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by GoldenSpade**
[JS-file](https://github.com/BilliZane/js_inverse_robot/blob/develop/src/inverseRobot.js)

### CODE REVIEW

**Review by stas-dukach — APPROVED** [review]
good

File: src/inverseRobot.js [hunk 1]
```diff
@@ -24,7 +24,28 @@
...
+    mainPropArray.push(key, robot[key]);
+  }
+
+  mainPropArray.sort();
+
+  for (let i = 0; i < mainPropArray.length; i++) {
+    if (mainPropArray[i] === mainPropArray[i + 1]) {
+      return null;
+    }
+  }
```
Comment (VinnieJ-2k20) [inline src/inverseRobot.js:40]: There are much more easier ways to check if an object has certain property.
Use `hasOwnProperty` here.

File: src/inverseRobot.js [hunk 2]
```diff
@@ -24,7 +24,28 @@
...
+
+  Object.keys(robot).forEach(function(value) {
+    const key = robot[value];
```
Comment (VinnieJ-2k20) [inline src/inverseRobot.js:43]: It's better to get used to arrow function from the start.

```suggestion
  Object.keys(robot).forEach((value) => {
    const key = robot[value];
```

**Review by VinnieJ-2k20 — CHANGES_REQUESTED** [review]
You need to optimize your code.
Details in comments.

File: src/inverseRobot.js [hunk 3]
```diff
@@ -23,8 +23,18 @@
...
  * @return {object}
  */
-function inverseRobot(robot) {
-  // write code here
-}
+
+const inverseRobot = (robot) => {
+  const robotReverse = {};
+
+  for (const key of Object.keys(robot)) {
```
Comment (varseniuk) [inline src/inverseRobot.js:30]:
```suggestion
  for (const key in robot) {
```
You could use `for`-`in` loop here

**Review by varseniuk — APPROVED** [review]
Good! 👍 
Left one comment

**Review by mykola-yarchuk — APPROVED** [review]

**Pull request closed without merging**